uni/lib/view/exams/exams.dart

```diff
@@ -19,55 +20,64 @@ class ExamsPageView extends StatefulWidget {
   State<StatefulWidget> createState() => ExamsPageViewState();
 }
 
-/// Tracks the state of `ExamsLists`.
 class ExamsPageViewState extends GeneralPageViewState<ExamsPageView> {
-  final double borderRadius = 10;
+  List<String> hiddenExams = PreferencesController.getHiddenExams();
+  Map<String, bool> filteredExamTypes =
+      PreferencesController.getFilteredExams();
 
   @override
   Widget getBody(BuildContext context) {
-    return LazyConsumer<ExamProvider>(
-      builder: (context, examProvider) {
+    return LazyConsumer<ExamProvider, List<Exam>>(
+      builder: (context, exams) {
         return ListView(
           children: <Widget>[
+            ExamPageTitle(
+              () => setState(() {
+                filteredExamTypes = PreferencesController.getFilteredExams();
+              }),
```

Comment:
I don't like this at all, but I see why.
I don't say now, but why don't we create a provider for Preferences and these components watch the preferences?

Reply:
Funny enough, I also think it is a bit weird to do this manually. I've looked into https://pub.dev/packages/rx_shared_preferences but it does not seem to be maintained. I suggest implementing it ourselves in a separate PR so this doesn't get even more polluted. Feel free to open an issue